Mathlib/CategoryTheory/DifferentialObject.lean

```diff
@@ -0,0 +1,368 @@
+
+variable {C}
+
+namespace DifferentialObject
+
+@[simp]
+lemma d_squared (X : DifferentialObject C) : X.d ≫ X.d⟦1⟧' = 0 := X.d_squared'
+
+/-- A morphism of differential objects is a morphism commuting with the differentials. -/
+@[ext] -- Porting note: Removed `nolint has_nonempty_instance`
+structure Hom (X Y : DifferentialObject C) where
+  /-- the morphism between underlying objects of the two differentiable object -/
+  f : X.pt ⟶ Y.pt
+  /-- the morphism should form a commutative square:
+  ```
+     d
+  X ---> X
+  | f    | f
+  v      v
+  Y ---> Y
+     d
+  ```
+  -/
+  comm' : X.d ≫ f⟦1⟧' = f ≫ Y.d := by aesop_cat -- Porting note: Originally `by obviously`
+#align category_theory.differential_object.hom CategoryTheory.DifferentialObject.Hom
+
+-- Porting note : this resulted in a `INCORRECT DEF/LEMMA` liner error, so `Hom.comm` is
+-- made manually
+-- restate_axiom Hom.comm'
+-- attribute [reassoc (attr := simp)] Hom.comm
+
+namespace Hom
+
+@[simp, reassoc (attr := simp)]
+lemma comm {X Y : DifferentialObject C} (f : Hom X Y) : X.d ≫ f.f⟦1⟧' = f.f ≫ Y.d := f.comm'
+
+/-- The identity morphism of a differential object. -/
+@[simps]
+def id (X : DifferentialObject C) : Hom X X where
+  f := 𝟙 X.pt
+#align category_theory.differential_object.hom.id CategoryTheory.DifferentialObject.Hom.id
+
+/-- The composition of morphisms of differential objects. -/
+@[simps]
+def comp {X Y Z : DifferentialObject C} (f : Hom X Y) (g : Hom Y Z) : Hom X Z where
+  f := f.f ≫ g.f
+#align category_theory.differential_object.hom.comp CategoryTheory.DifferentialObject.Hom.comp
+
+end Hom
+
+instance categoryOfDifferentialObjects : Category (DifferentialObject C) where
+  Hom := Hom
+  id := Hom.id
+  comp f g := Hom.comp f g
+#align category_theory.differential_object.category_of_differential_objects CategoryTheory.DifferentialObject.categoryOfDifferentialObjects
+
+@[simp]
+theorem id_f (X : DifferentialObject C) : (𝟙 X : X ⟶ X).f = 𝟙 X.pt := rfl
+#align category_theory.differential_object.id_f CategoryTheory.DifferentialObject.id_f
+
+@[simp]
+theorem comp_f {X Y Z : DifferentialObject C} (f : X ⟶ Y) (g : Y ⟶ Z) : (f ≫ g).f = f.f ≫ g.f := rfl
+#align category_theory.differential_object.comp_f CategoryTheory.DifferentialObject.comp_f
+
+@[simp]
+theorem eqToHom_f {X Y : DifferentialObject C} (h : X = Y) :
+    Hom.f (eqToHom h) = eqToHom (congr_arg _ h) := by
+  subst h
+  rw [eqToHom_refl, eqToHom_refl]
+  rfl
+#align category_theory.differential_object.eq_to_hom_f CategoryTheory.DifferentialObject.eqToHom_f
+
+variable (C)
+
+/-- The forgetful functor taking a differential object to its underlying object. -/
+def forget : DifferentialObject C ⥤ C where
+  obj X := X.pt
+  map f := f.f
+#align category_theory.differential_object.forget CategoryTheory.DifferentialObject.forget
+
+instance forget_faithful : Faithful (forget C) where
+  map_injective := Hom.ext _ _ -- Porting note: Added a definition for `map_injective`
```

Comment:
The problem is that you need to restate the `ext` lemma for `⟶` rather than `Hom`. Then `aesop_cat` will be able to fill in this field, as before.